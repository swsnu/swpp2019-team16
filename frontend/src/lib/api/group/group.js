import client from '../client';

export function createGroup({ groupId, riders, driver, from, to }) {
  return client.post(`/api/v1/group`, { groupId, riders, driver, from, to });
}

export function acceptGroup({ groupId, driverId }) {
  return client.put(`/api/v1/group/${groupId}`, driverId);
}

export function onTaxi(riderId) {
  return client.put(`/api/v1/group/taxi/riders`, riderId);
}

export function departure({ groupId, departureTime }) {
  return client.put(`/api/v1/group/taxi/departure`, { groupId, departureTime });
}

export function fetchMyGroup(groupId) {
  return client.get(`/api/v1/group/${groupId}`);
}

export function notifyDriverLocation({ driverId, groupId, locationInfo }) {
  return client.put(`/api/v1/group/taxi/location`, {
    driverId,
    groupId,
    locationInfo,
  });
}

export function confirmCost({ groupId, totalCost }) {
  return client.put(`/api/v1/group/cost`, { groupId, totalCost });
}
