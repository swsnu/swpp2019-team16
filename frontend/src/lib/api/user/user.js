import client from '../client';

export function get({ id }) {
  return client.get(`/api/v1/user/${id}`);
}

export function updatePoint({ userId, point }) {
  return client.put(`/api/v1/user/${userId}`, { point });
}

export function riderOnTaxi({ riderId }) {
  return client.put(`/api/v1/user/taxi/rider/${riderId}`);
}

export function driverGoTaxi({ driverId }) {
  return client.put(`/api/v1/user/taxi/driver/${driverId}`);
}
