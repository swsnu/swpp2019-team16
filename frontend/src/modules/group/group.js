import { createAction, handleActions } from 'redux-actions';
import { takeLatest } from 'redux-saga/effects';
import * as groupAPI from '../../lib/api/group/group';
import createRequestSaga, {
  createRequestActionTypes,
} from '../../lib/createRequestSaga';

const [
  ACCEPT_GROUP,
  ACCEPT_GROUP_SUCCESS,
  ACCEPT_GROUP_FAILURE,
] = createRequestActionTypes('group/ACCEPT_GROUP');

const [ON_TAXI, ON_TAXI_SUCCESS, ON_TAXI_FAILURE] = createRequestActionTypes(
  'group/ON_TAXI',
);

const [
  DEPARTURE,
  DEPARTURE_SUCCESS,
  DEPARTURE_FAILURE,
] = createRequestActionTypes('group/DEPARTURE');

const [
  FETCH_MY_GROUP,
  FETCH_MY_GROUP_SUCCESS,
  FETCH_MY_GROUP_FAILURE,
] = createRequestActionTypes('group/FETCH_MY_GROUP');

const [
  NOTIFY_DRIVER_LOCATION,
  NOTIFY_DRIVER_LOCATION_SUCCESS,
  NOTIFY_DRIVER_LOCATION_FAILURE,
] = createRequestActionTypes('group/NOTIFY_DRIVER_LOCATION');

const [
  CONFIRM_COST,
  CONFIRM_COST_SUCCESS,
  CONFIRM_COST_FAILURE,
] = createRequestActionTypes('group/CONFIRM_COST');

export const GROUP_CREATED = 'group/GROUP_CREATED';
export const GROUP_UPDATED = 'group/GROUP_UPDATED';
export const UNLOAD_GROUP = 'group/UNLOAD_GROUP';
export const GROUP_COST_UPDATED = 'group/GROUP_COST_UPDATED';
export const INIT_COST_CONFIRMED = 'group/INIT_COST_CONFIRMED';

export const groupCreated = createAction(
  GROUP_CREATED,
  ({ groupId, from, to }) => ({
    groupId,
    from,
    to,
  }),
);

export const groupCostUpdated = createAction(
  GROUP_COST_UPDATED,
  ({ groupId, riderCost, cost }) => ({
    groupId,
    riderCost,
    cost,
  }),
);

export const groupUpdated = createAction(
  GROUP_UPDATED,
  ({ groupId, driver, riders, fromLocation, toLocation }) => ({
    groupId,
    driver,
    riders,
    from: fromLocation,
    to: toLocation,
  }),
);

export const unloadGroup = createAction(UNLOAD_GROUP);

export const acceptGroup = createAction(
  ACCEPT_GROUP,
  ({ groupId, driverId }) => ({
    groupId,
    driverId,
  }),
);
export const onTaxi = createAction(ON_TAXI, riderId => riderId);
export const departure = createAction(
  DEPARTURE,
  ({ groupId, departureTime }) => ({
    groupId,
    departureTime,
  }),
);
export const fetchMyGroup = createAction(FETCH_MY_GROUP, groupId => groupId);
export const notifyDriverLocation = createAction(
  NOTIFY_DRIVER_LOCATION,
  ({ driverId, groupId, locationInfo }) => ({
    driverId,
    groupId,
    locationInfo,
  }),
);
export const confirmCost = createAction(
  CONFIRM_COST,
  ({ groupId, totalCost }) => ({
    groupId,
    totalCost,
  }),
);
export const initCostConfirmed = createAction(INIT_COST_CONFIRMED);

export const acceptGroupSaga = createRequestSaga(
  ACCEPT_GROUP,
  groupAPI.acceptGroup,
);
export const onTaxiSaga = createRequestSaga(ON_TAXI, groupAPI.onTaxi);
export const departureSaga = createRequestSaga(DEPARTURE, groupAPI.departure);
export const fetchMyGroupSaga = createRequestSaga(
  FETCH_MY_GROUP,
  groupAPI.fetchMyGroup,
);
export const notifyDriverLocationSaga = createRequestSaga(
  NOTIFY_DRIVER_LOCATION,
  groupAPI.notifyDriverLocation,
);
export const confirmCostSaga = createRequestSaga(
  CONFIRM_COST,
  groupAPI.confirmCost,
);

export function* groupSaga() {
  yield takeLatest(ACCEPT_GROUP, acceptGroupSaga);
  yield takeLatest(ON_TAXI, onTaxiSaga);
  yield takeLatest(DEPARTURE, departureSaga);
  yield takeLatest(FETCH_MY_GROUP, fetchMyGroupSaga);
  yield takeLatest(NOTIFY_DRIVER_LOCATION, notifyDriverLocationSaga);
  yield takeLatest(CONFIRM_COST, confirmCostSaga);
}

const initialState = {
  group: null,
  error: null,
  costConfirmed: false,
};

const group = handleActions(
  {
    [GROUP_CREATED]: (state, { payload: group }) => ({
      ...state,
      group,
    }),

    [GROUP_UPDATED]: (state, { payload: group }) => ({
      ...state,
      group,
    }),

    [UNLOAD_GROUP]: state => ({
      ...state,
      group: null,
    }),

    [GROUP_COST_UPDATED]: (state, { payload: group }) => ({
      ...state,
      group: {
        ...state.group,
        ...group,
      },
    }),

    [ACCEPT_GROUP_SUCCESS]: (state, { payload: group }) => ({
      ...state,
      group: group,
    }),

    [ACCEPT_GROUP_FAILURE]: (state, { payload: error }) => ({
      ...state,
      error: error,
    }),

    [ON_TAXI_SUCCESS]: (state, { payload: group }) => ({
      ...state,
      group: group,
    }),

    [ON_TAXI_FAILURE]: (state, { payload: error }) => ({
      ...state,
      error: error,
    }),

    [DEPARTURE_SUCCESS]: (state, { payload: group }) => ({
      ...state,
      group: group,
    }),

    [DEPARTURE_FAILURE]: (state, { payload: error }) => ({
      ...state,
      error: error,
    }),

    [FETCH_MY_GROUP_SUCCESS]: (state, { payload: group }) => ({
      ...state,
      group: group,
    }),

    [FETCH_MY_GROUP_FAILURE]: (state, { payload: error }) => ({
      ...state,
      error: error,
    }),

    [NOTIFY_DRIVER_LOCATION_SUCCESS]: (state, { payload: group }) => ({
      ...state,
      group: group,
    }),
    [NOTIFY_DRIVER_LOCATION_FAILURE]: (state, { payload: error }) => ({
      ...state,
      error: error,
    }),

    [CONFIRM_COST_SUCCESS]: (state, { payload: group }) => ({
      ...state,
      group: {
        ...state.group,
        ...group,
      },
      error: null,
      costConfirmed: true,
    }),

    [CONFIRM_COST_FAILURE]: (state, { payload: error }) => ({
      ...state,
      error: error,
    }),
    [INIT_COST_CONFIRMED]: state => ({
      ...state,
      costConfirmed: initialState.costConfirmed,
    }),
  },
  initialState,
);

export default group;
