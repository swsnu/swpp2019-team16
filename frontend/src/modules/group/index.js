import group from './group';

export * from './group';
export default group;