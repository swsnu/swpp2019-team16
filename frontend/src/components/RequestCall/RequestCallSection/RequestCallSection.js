import React, { useCallback, useState } from 'react';
import Button from '../../common/Button';
import PropTypes from 'prop-types';
import recognizer from '../../../lib/azure';
import Checkbox from '../../common/Checkbox';
import { makeStyles } from '@material-ui/styles';
import { UserPropsTypes } from '../../../types/user';
import { Typography } from '@material-ui/core';
import Map from '../../common/Map';
import MapPin from '../../common/MapPin';
import { GroupPropTypes } from '../../../types/group';
import Heading from '../../common/Heading';

const useStyles = makeStyles({
  root: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    padding: '1rem',
  },
  title: {
    display: 'flex',
    justifyContent: 'center',
    marginBottom: '1rem',
  },
  map: {
    marginBottom: '1rem',
  },
  location: {
    display: 'flex',
    marginBottom: '1rem',
    justifyContent: 'center',
  },
  button: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
  },
});

RequestCallSection.propTypes = {
  user: UserPropsTypes,
  group: GroupPropTypes,
  onClickRequestCall: PropTypes.func.isRequired,
};

function RequestCallSection({ user, group, onClickRequestCall }) {
  const styles = useStyles();
  const [speechToText, setSpeechToText] = useState(false);
  const triggerText = 'Stop';

  // TODO: add test case
  const onSTTHandler = useCallback(() => {
    if (!speechToText) {
      console.log('Active STT');
      recognizer.recognized = (r, event) => {
        console.log('Recognized message: ' + event.result.text);
        if (event.result.text.includes(triggerText)) {
          console.log('Triggered message: ' + triggerText);
          recognizer.stopContinuousRecognitionAsync();
          onClickRequestCall({
            groupId: group.groupId,
            driverId: user.id,
          });
        }
      };
      recognizer.startContinuousRecognitionAsync();
    } else {
      console.log('Deactive STT');
      recognizer.stopContinuousRecognitionAsync();
    }
    setSpeechToText(!speechToText);
  }, [group, user, speechToText, onClickRequestCall]);

  return (
    <div className={styles.root}>
      <div className={styles.title}>
        <Heading
          title={
            group
              ? "There's new carpool group!"
              : 'We are waiting for carpool group'
          }
        />
      </div>
      <div>
        <Checkbox
          value={'STT Mode'}
          name={'STT'}
          onClick={onSTTHandler}
          checked={speechToText}
        />
      </div>
      {group && (
        <div>
          <div className={styles.map}>
            <Map
              zoom={15}
              center={{
                lat: 37.480126,
                lng: 126.952436,
              }}
              width={'100%'}
              height={'250px'}
            >
              <MapPin lat={37.480126} lng={126.952436} />
              <MapPin lat={37.477023} lng={126.961957} />
            </Map>
          </div>
          <div className={styles.location}>
            <Typography variant={'h3'}>
              {group.from} ~ {group.to}
            </Typography>
          </div>

          <div className={styles.button}>
            <Button
              id="request-call-submit-button"
              variant="contained"
              color="primary"
              onClick={() =>
                onClickRequestCall({
                  groupId: group.groupId,
                  driverId: user.id,
                })
              }
              children="Accept request"
            />
          </div>
        </div>
      )}
    </div>
  );
}

export default RequestCallSection;
