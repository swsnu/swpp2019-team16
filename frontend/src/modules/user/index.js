import user from './user';

export * from './user';
export default user;
