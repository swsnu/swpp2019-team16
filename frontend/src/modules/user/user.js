import { createAction, handleActions } from 'redux-actions';
import { takeLatest, put } from 'redux-saga/effects';
import * as userAPI from '../../lib/api/user';
import createRequestSaga, {
  createRequestActionTypes,
} from '../../lib/createRequestSaga';

const TEMP_SET_USER = 'user/TEMP_SET_USER';
const SOMEONE_ON_TAXI = 'user/SOMEONE_ON_TAXI';
const GO_TAXI = 'user/GO_TAXI';
const INIT_POINT_UPDATED_FLAG = 'user/INIT_POINT_UPDATED_FLAG';
const [CHECK, CHECK_SUCCESS, CHECK_FAILURE] = createRequestActionTypes(
  'user/CHECK',
);
const [LOGOUT, LOGOUT_SUCCESS, LOGOUT_FAILURE] = createRequestActionTypes(
  'user/LOGOUT',
);
const [
  UPDATE_POINT,
  UPDATE_POINT_SUCCESS,
  UPDATE_POINT_FAILURE,
] = createRequestActionTypes('user/UPDATE_POINT');
const [
  RIDER_ON_TAXI,
  RIDER_ON_TAXI_SUCCESS,
  RIDER_ON_TAXI_FAILURE,
] = createRequestActionTypes('user/RIDER_ON_TAXI');
const [
  DRIVER_GO_TAXI,
  DRIVER_GO_TAXI_SUCCESS,
  DRIVER_GO_TAXI_FAILURE,
] = createRequestActionTypes('user/DRIVER_GO_TAXI');

export const tempSetUser = createAction(TEMP_SET_USER, user => user);
export const check = createAction(CHECK, ({ id }) => ({ id }));
export const logout = createAction(LOGOUT);
export const updatePoint = createAction(UPDATE_POINT, ({ userId, point }) => ({
  userId,
  point,
}));

export const checkSaga = createRequestSaga(CHECK, userAPI.get);
export const riderOnTaxi = createAction(RIDER_ON_TAXI, ({ riderId }) => ({
  riderId,
}));
export const goTaxi = createAction(DRIVER_GO_TAXI, ({ driverId }) => ({
  driverId,
}));
export const someoneOnTaxi = createAction(SOMEONE_ON_TAXI, ({ riderId }) => ({
  riderId,
}));
export const taxiGone = createAction(GO_TAXI);
export const initPointUpdatedFlag = createAction(INIT_POINT_UPDATED_FLAG);

const riderOnTaxiSaga = createRequestSaga(RIDER_ON_TAXI, userAPI.riderOnTaxi);
const driverGoTaxiSaga = createRequestSaga(
  DRIVER_GO_TAXI,
  userAPI.driverGoTaxi,
);

function checkFailureSaga() {
  try {
    localStorage.removeItem('user');
  } catch (e) {
    console.error('localStorage is not working');
  }
}

export function* logoutSaga() {
  try {
    localStorage.removeItem('user');
    yield put({ type: LOGOUT_SUCCESS });
  } catch (e) {
    console.error(e);
    yield put({ type: LOGOUT_FAILURE });
  }
}

export const updatePointSaga = createRequestSaga(
  UPDATE_POINT,
  userAPI.updatePoint,
);
export function* userSaga() {
  yield takeLatest(CHECK, checkSaga);
  yield takeLatest(CHECK_FAILURE, checkFailureSaga);
  yield takeLatest(LOGOUT, logoutSaga);
  yield takeLatest(UPDATE_POINT, updatePointSaga);
  yield takeLatest(RIDER_ON_TAXI, riderOnTaxiSaga);
  yield takeLatest(DRIVER_GO_TAXI, driverGoTaxiSaga);
}

const initialState = {
  user: null,
  checkError: null,
  logoutError: null,
  updatePointError: null,
  onTaxiRidersList: [],
  driverGoTaxi: false,
  onTaxiError: null,
  goTaxiError: null,
  pointUpdated: false,
};

const user = handleActions(
  {
    [TEMP_SET_USER]: (state, { payload: user }) => ({
      ...state,
      user,
    }),
    [CHECK_SUCCESS]: (state, { payload: user }) => ({
      ...state,
      user,
      checkError: null,
    }),
    [CHECK_FAILURE]: (state, { payload: error }) => ({
      ...state,
      user: null,
      checkError: error,
    }),
    [LOGOUT_SUCCESS]: state => ({
      ...state,
      user: null,
      logoutError: null,
    }),
    [LOGOUT_FAILURE]: (state, { payload: error }) => ({
      ...state,
      logoutError: error,
    }),
    [UPDATE_POINT_SUCCESS]: (state, { payload: user }) => ({
      ...state,
      user: {
        ...user,
        user,
      },
      pointUpdated: true,
      updatePointError: null,
    }),
    [UPDATE_POINT_FAILURE]: (state, { payload: error }) => ({
      ...state,
      updatePointError: error,
    }),
    [RIDER_ON_TAXI_FAILURE]: (state, { payload: error }) => ({
      ...state,
      onTaxiError: error,
    }),
    [DRIVER_GO_TAXI_FAILURE]: (state, { payload: error }) => ({
      ...state,
      goTaxiError: error,
    }),
    [SOMEONE_ON_TAXI]: (state, { payload: riderId }) => ({
      ...state,
      onTaxiRidersList: [...state.onTaxiRidersList, riderId],
    }),
    [GO_TAXI]: state => ({
      ...state,
      driverGoTaxi: true,
    }),
    [INIT_POINT_UPDATED_FLAG]: state => ({
      ...state,
      pointUpdated: initialState.pointUpdated,
    }),
  },
  initialState,
);

export default user;
