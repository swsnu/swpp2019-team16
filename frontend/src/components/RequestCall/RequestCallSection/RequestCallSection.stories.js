import React from 'react';
import { BrowserRouter } from 'react-router-dom';

import { storiesOf } from '@storybook/react';

import RequestCallSection from './RequestCallSection';
import { ThemeProvider } from '@material-ui/styles';
import theme from '../../../lib/styles/theme';
import { CssBaseline } from '@material-ui/core';
import { mockDriver } from '../../../types/__mock__/user';

storiesOf('RequestCall/RequestCallSection', module)
  .addDecorator(story => (
    <BrowserRouter>
      <ThemeProvider theme={theme}>
        <CssBaseline />
        {story()}
      </ThemeProvider>
    </BrowserRouter>
  ))
  .add('default', () => (
    <RequestCallSection
      groupId={1}
      user={mockDriver}
      fromLocation={'SNU Station'}
      toLocation={'301 Building'}
      onClickRequestCall={() => {}}
    />
  ));
