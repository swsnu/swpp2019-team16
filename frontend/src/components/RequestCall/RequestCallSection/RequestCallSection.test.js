import React from 'react';
import { render } from '@testing-library/react';
import RequestCallSection from './RequestCallSection';
import { mockDriver } from 'types/__mock__/user';
import { mockGroup } from 'types/__mock__/group';

describe('RequestCall />', () => {
  it('SHOULD match with snapshot WHEN all data in', async () => {
    const { container } = render(
      <RequestCallSection
        user={mockDriver}
        group={mockGroup}
        onClickRequestCall={() => {}}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
