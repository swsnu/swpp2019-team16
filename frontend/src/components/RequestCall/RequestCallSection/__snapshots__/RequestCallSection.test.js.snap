// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestCall /> SHOULD match with snapshot WHEN all data in 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="makeStyles-title-2"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2"
        title="There's new carpool group!"
      >
        There's new carpool group!
      </h2>
    </div>
    <div>
      <div>
        <label
          class="MuiFormControlLabel-root"
          data-testid="STT Mode"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-48 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-51"
                data-indeterminate="false"
                name="STT"
                type="checkbox"
                value="STT Mode"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            STT Mode
          </span>
        </label>
      </div>
    </div>
    <div>
      <div
        class="makeStyles-map-3"
      >
        <div
          class="makeStyles-root-73 makeStyles-root-74"
        >
          <div
            style="width: 100%; height: 100%; margin: 0px; padding: 0px; position: relative;"
          >
            <div
              style="width: 100%; height: 100%; left: 0px; top: 0px; margin: 0px; padding: 0px; position: absolute;"
            />
            <div
              style="width: 50%; height: 50%; left: 50%; top: 50%; margin: 0px; padding: 0px; position: absolute;"
            >
              <div
                style="width: 100%; height: 100%; left: 0px; top: 0px; margin: 0px; padding: 0px; position: absolute;"
              >
                <div
                  style="width: 0px; height: 0px; left: 0px; top: 0px; background-color: transparent; position: absolute;"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root makeStyles-root-75"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root makeStyles-icon-76 MuiSvgIcon-colorSecondary"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2c-4.97 0-9 4.03-9 9 0 4.17 2.84 7.67 6.69 8.69L12 22l2.31-2.31C18.16 18.67 21 15.17 21 11c0-4.97-4.03-9-9-9zm0 2c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.3c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <div
                  style="width: 0px; height: 0px; left: 221.85537991020828px; top: 91.11249785777181px; background-color: transparent; position: absolute;"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root makeStyles-root-75"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root makeStyles-icon-76 MuiSvgIcon-colorSecondary"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2c-4.97 0-9 4.03-9 9 0 4.17 2.84 7.67 6.69 8.69L12 22l2.31-2.31C18.16 18.67 21 15.17 21 11c0-4.97-4.03-9-9-9zm0 2c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.3c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="makeStyles-location-4"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3"
        >
           ~ 
        </h3>
      </div>
      <div
        class="makeStyles-button-5"
      >
        <button
          class="MuiButtonBase-root MuiButton-root makeStyles-root-77 makeStyles-root-80 MuiButton-contained makeStyles-contained-78 makeStyles-contained-81 MuiButton-contained makeStyles-contained-78 makeStyles-contained-81"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Accept request
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
